client/state/posts/reducer.js

```diff
@@ -261,7 +263,16 @@ export function edits( state = {}, action ) {
 				}
 			} );
 
-		case POST_EDITS_RESET:
+		case EDITOR_START:
+			return Object.assign( {}, state, {
+				[ action.siteId ]: {
+					...state[ action.siteId ],
+					[ action.postId || '' ]: { type: action.postType }
+				}
+			} );
+
+		case EDITOR_STOP:
+		case POST_SAVE_SUCCESS:
```

Comment:
About that, I think we should move the `type` out of the `edits` state. I don't think it should be stored there since it's not something editable. I would store this on the `ui` subtree, the same way we store the current edited ID. Maybe another PR